client/src/client.rs

```diff
@@ -148,6 +153,7 @@ pub struct Client<R: Reactor<Publisher>> {
     events: event::Subscriber<Event>,
     blocks: event::Subscriber<(Block, Height)>,
     filters: event::Subscriber<(BlockFilter, BlockHash, Height)>,
+    transactions: Arc<RwLock<HashMap<Txid, (Transaction, HashSet<PeerId>, Height)>>>,
```

Comment:
This should be owned by the `TxnManager`, then you don't need a lock.

Reply:
Hmmm, which means all storage is dropped if the TxnManager is exited. I assume transaction storage is lifetime so that we can also implement rebroadcast.